bin/ch/WScriptJsrt.cpp

```diff
@@ -37,6 +38,8 @@
 
 MessageQueue* WScriptJsrt::messageQueue = nullptr;
 std::map<std::string, JsModuleRecord>  WScriptJsrt::moduleRecordMap;
+std::map<JsModuleRecord, std::string>  WScriptJsrt::moduleDirMap;
```

Comment:
Adding critical sections to lock global state updates in these hosts.
Confirmed with @bterlson that the lifetime of JsModuleRecord is tied to the Realm.
Yes, will have to figure out a way to clear moduleDirMap once the module record gets collected.

---
In reply to: [125369988](https://github.com/Microsoft/ChakraCore/pull/3284#discussion_r125369988) [](ancestors = 125369988)